libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/PropertyMocks/MockSettingsMiddleware.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Testing.PropertyMocks
+{
+    /// <summary>
+    /// Middleware used for mocking settings properties.
+    /// </summary>
+    public class MockSettingsMiddleware : IMiddleware
+    {
+        private readonly string prefix = $"{ScopePath.Settings}.";
+        private readonly Dictionary<string, string> mockData = new Dictionary<string, string>();
+
+        public MockSettingsMiddleware(List<PropertyMock> properties)
+        {
+            foreach (var property in properties)
+            {
+                if (property is PropertiesMock mock)
+                {
+                    foreach (var assignment in mock.Assignments)
+                    {
+                        if (assignment.Property.StartsWith(prefix, StringComparison.Ordinal))
+                        {
+                            if (assignment.Value is string value)
+                            {
+                                var path = assignment.Property.Substring(prefix.Length);
+
+                                if (!mockData.ContainsKey(path))
+                                {
+                                    // Note that settings use : as separator in ConfigurationBuilder.
+                                    mockData.Add(path.Replace('.', ':'), value);
+                                }
+                            }
+                            else
+                            {
+                                throw new NotSupportedException($"Only string is supported as value for mocking settings. {assignment.Value} is not supported.");
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default)
+        {
+            if (mockData.Count != 0)
+            {
+                IConfigurationBuilder configBuilder = new ConfigurationBuilder();
+                var previousConfig = turnContext.TurnState.Get<IConfiguration>();
+                if (previousConfig != null)
+                {
+                    configBuilder = configBuilder.AddConfiguration(previousConfig);
+                }
+
+                configBuilder.AddInMemoryCollection(mockData);
```

Comment:
this can be moved to constructor

Reply:
Resolved here https://github.com/microsoft/botbuilder-dotnet/pull/4226
Well, I did not setup it in constructor for broader usage like in TestScript.ExecuteAsync, use an existing TestAdapter.